web/src/pages/app/mods/SideBar/index.tsx

```diff
@@ -27,6 +28,15 @@ export default function SideBar() {
   const navigate = useNavigate();
   const { currentApp, setCurrentPage, userInfo } = useGlobalStore();
 
+  const blockLeavePage = useCallback((pageId: any) => {
+    if (window.location.pathname.includes("/function") && pageId !== "function") {
+      // eslint-disable-next-line no-alert
+      return !window.confirm(String(t("FunctionPanel.LeaveFunctionPage")));
+    } else {
+      return false;
+    }
+  }, []);
+
```

Comment:
only the pageId is 'function' and the user want to close the browser tab, then show the confirm dialog.